core/include/grids/populator.hpp

```diff
@@ -80,7 +83,10 @@ namespace detray
      * 
      * @note bare_value and store_value are different in this case
      **/
-    template <typename value_type, value_type kInvalid, unsigned int kDIM, bool kSORT = false>
+    template <unsigned int kDIM, 
```

Comment:
Since the grid is templated on the populator type, is it possible to remove the kDIM template here?

Reply:
I guess so, will have a look.

Reply:
This is needed, it's the dimension of the internal grid array storage, i.e. if you want to store two objects per grid entry.

Reply:
Right. So I guess there wont be too many different types of those grids in a detector and we can set the dimension in a grid type for the detector class?

Reply:
For a double-sided detector,  (like SCT ATLAS, etc.) you would like to have 2 entries per bin, but it could also be differently, but fixed size. The completer makes sure a maximum of the fixed size entries are used, if you want dynamic size, you need to take the attacher.